v2/tools/generator/internal/astbuilder/builder.go

```diff
@@ -255,6 +255,30 @@ func ReturnIfExpr(cond dst.Expr, returns ...dst.Expr) *dst.IfStmt {
 	}
 }
 
+// CheckIfNotNilAndAddStatements checks if the ident is non-nil, and if it is executes the provided statements.
+//
+//	if <ident> != nil {
+//		<stmt>
+//	}
+func CheckIfNotNilAndAddStatements(ident string, stmts ...dst.Stmt) dst.Stmt {
```

Comment:
I think we already have `IfNotNil` which does basically exactly this?

Reply:
we have `CheckErrorAndSingleStatement(stmt dst.Stmt) dst.Stmt` which checks the `err`.